beacon-chain/core/blocks/validity_conditions.go

```diff
@@ -79,3 +82,25 @@ func IsSlotValid(slot uint64, genesisTime time.Time) bool {
 	}
 	return isValid
 }
+
+// DepositsToRemove returns the deposits that we have to remove.
+func DepositsToRemove() []*pb.Deposit {
+	deps := make([]*pb.Deposit, len(depositsToRemove))
+	copy(deps, depositsToRemove)
+	return deps
+}
+
+// ClearFromDepositRemovalList removes the deposit from the list.
+func ClearFromDepositRemovalList(invalidDeposit *pb.Deposit) {
+	for i, deposit := range depositsToRemove {
+		if proto.Equal(deposit, invalidDeposit) {
+			depositsToRemove = append(depositsToRemove[:i], depositsToRemove[i+1:]...)
+		}
+	}
+}
+
+func addToDepositRemovalList(removalEnabled bool, deposit *pb.Deposit) {
+	if removalEnabled {
```

Comment:
hmm, feels weird to use `removalEnabled` here, better if we use it in block operations?